File: ChessApp.Core/Enums/OnlineTimeControl.cs
namespace ChessApp.Core.Enums;

public enum OnlineTimeControl
{
    Blitz3 = 0,         // 3+0
    Blitz3Plus2 = 1,    // 3+2
    Blitz5 = 2,         // 5+0
    Blitz5Plus2 = 3,    // 5+2
    Rapid10 = 4,        // 10+0
    Rapid10Plus2 = 5,   // 10+2
    Rapid15 = 6,        // 15+0
    Rapid15Plus3 = 7,   // 15+3
    Classical30 = 8,    // 30+0
}

public static class OnlineTimeControlExtensions
{
    public static (int InitialSeconds, int IncrementSeconds) ToSpec(this OnlineTimeControl tc)
        => tc switch
        {
            OnlineTimeControl.Blitz3 => (180, 0),
            OnlineTimeControl.Blitz3Plus2 => (180, 2),
            OnlineTimeControl.Blitz5 => (300, 0),
            OnlineTimeControl.Blitz5Plus2 => (300, 2),
            OnlineTimeControl.Rapid10 => (600, 0),
            OnlineTimeControl.Rapid10Plus2 => (600, 2),
            OnlineTimeControl.Rapid15 => (900, 0),
            OnlineTimeControl.Rapid15Plus3 => (900, 3),
            OnlineTimeControl.Classical30 => (1800, 0),
            _ => (300, 0),
        };

    public static string ToPgnTag(this OnlineTimeControl tc)
    {
        var (init, inc) = tc.ToSpec();
        return $"{init}+{inc}";
    }
}

File: chess_app/lib/features/online_game/logic/online_time_control.dart
enum OnlineTimeControl {
  blitz3(0, 180, 0, '3+0', 'Blitz'),
  blitz3Plus2(1, 180, 2, '3+2', 'Blitz'),
  blitz5(2, 300, 0, '5+0', 'Blitz'),
  blitz5Plus2(3, 300, 2, '5+2', 'Blitz'),
  rapid10(4, 600, 0, '10+0', 'Rapid'),
  rapid10Plus2(5, 600, 2, '10+2', 'Rapid'),
  rapid15(6, 900, 0, '15+0', 'Rapid'),
  rapid15Plus3(7, 900, 3, '15+3', 'Rapid'),
  classical30(8, 1800, 0, '30+0', 'Classical');

  final int wireValue;
  final int initialSeconds;
  final int incrementSeconds;
  final String label;
  final String category;

  const OnlineTimeControl(
    this.wireValue,
    this.initialSeconds,
    this.incrementSeconds,
    this.label,
    this.category,
  );

  Duration get initialDuration => Duration(seconds: initialSeconds);

  static OnlineTimeControl fromWire(int? v) {
    if (v == null) return OnlineTimeControl.blitz5;
    for (final tc in OnlineTimeControl.values) {
      if (tc.wireValue == v) return tc;
    }
    return OnlineTimeControl.blitz5;
  }
}
